### PR TITLE
12.0 mig report xml

### PR DESCRIPTION
**Pull request by ValentinVinagre**
Standard migration to 12.0

### CODE REVIEW

File: report_xml/views/webclient_templates.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <template id="report_xml.assets_backend" inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <script type="text/javascript" src="/report_xml/static/src/js/report/qwebactionmanager.js"/>
+        </xpath>
+
+    </template>
+</odoo>
```
Comment (etobella) [inline report_xml/views/webclient_templates.xml:9]: Missing new line

File: report_xml/tests/test_report_xml.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
```
Comment (etobella) [inline report_xml/tests/test_report_xml.py:1]: No need of this line on Python 3

File: report_xml/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014-2015  Grupo ESOC <www.grupoesoc.es>
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+from odoo.addons.web.controllers import main as report
+from odoo.http import content_disposition, route
+
+
+class ReportController(report.ReportController):
+    @route()
```
Comment (etobella) [inline report_xml/controllers/main.py:10]: Not blocking, but could you test this lines? Just in order to improve coverage.

**Comment by etobella** [comment]
@ValentinVinagre Can you attend the comments?

**Comment by pedrobaeza** [comment]
@ernestotejeda please review

File: report_xml/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014-2015  Grupo ESOC <www.grupoesoc.es>
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+{
+    "name": "XML Reports",
+    "version": "12.0.1.0.1",
```
Comment (ernestotejeda) [inline report_xml/__manifest__.py:7]:
```suggestion
    "version": "12.0.1.0.0",
```

File: report_xml/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
```
Comment (ernestotejeda) [inline report_xml/__manifest__.py:1]:
```suggestion
```

File: report_xml/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
```
Comment (ernestotejeda) [inline report_xml/__init__.py:1]:
```suggestion
```

File: report_xml/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
```
Comment (ernestotejeda) [inline report_xml/controllers/main.py:1]:
```suggestion
```

File: report_xml/controllers/__init__.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (ernestotejeda) [inline report_xml/controllers/__init__.py:1]:
```suggestion
```

File: report_xml/models/report_action.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
```
Comment (ernestotejeda) [inline report_xml/models/report_action.py:1]:
```suggestion
```

File: report_xml/static/src/js/report/qwebactionmanager.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+                    }
+                },
+            });
+            framework.unblockUI();
+            return
+        }
+        return self._super(action, options);
+    }
+});
+});
```
Comment (ernestotejeda) [inline report_xml/static/src/js/report/qwebactionmanager.js:41]:
```suggestion
});

```

File: report_xml/tests/__init__.py [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (ernestotejeda) [inline report_xml/tests/__init__.py:1]:
```suggestion
```

File: report_xml/models/report_action.py [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+_logger = logging.getLogger(__name__)
+
+
+class ReportAction(models.Model):
+    _inherit = "ir.actions.report"
+
+    report_type = fields.Selection(selection_add=[("qweb-xml", "XML")])
+
+    @api.model
+    def _get_report_from_name(self, report_name):
```
Comment (ernestotejeda) [inline report_xml/models/report_action.py:19]: I think it isn't necessary

**Review by ernestotejeda — COMMENTED** [review]
For some reason it isn't working in v12, would you check, please?

runbot reporting-engine v11
Peek 10-06-2019 10-00

runbot this migration:
Peek 10-06-2019 10-01

**Comment by pedrobaeza** [comment]
Superseeded by #295

**Pull request closed without merging**